### PR TITLE
[MIG] Old OpenERP document_ftp module to 8.0 migration

### PR DESCRIPTION
**Pull request by cmsalmeida**

### CODE REVIEW

**Comment by cmsalmeida** [comment]
But I've signed and payed the OCA
https://odoo-community.org/members/thinkopen-solutions-brasil-carlos-miguel-sousa-almeida-1927
What can be done?

**Comment by angelmoya** [comment]
Fix all pep8, and squash all commits.

Don't use import pooler.

**Comment by cmsalmeida** [comment]
WIP

**Comment by max3903** [comment]
@cmsalmeida We were missing your Github login. It is fixed now. 

Cheers,

**Comment by angelmoya** [comment]
Do you have planned to fix this PR? There are a lot of errors in code.

If you don't have time maybe I could do another PR.

File: document_ftp/ftpserver/__init__.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    PASSIVE_PORTS = None
+    pps = config.get('ftp_server_passive_ports', '').split(':')
+    if len(pps) == 2:
+        PASSIVE_PORTS = int(pps[0]), int(pps[1])
+
+    class FtpServer(threading.Thread):
+
+        def run(self):
+            autho = authorizer.Authorizer()
+            ftp_server.FTPHandler.Authorizer = autho
```
Comment (angelmoya) [inline document_ftp/ftpserver/__init__.py:57]: ftp_server > ftpserver this and the following

File: document_ftp/ftpserver/__init__.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import threading
+import logging
+
+from . import authorizer
+from . import abstracted_fs
+from . import ftp_server
```
Comment (angelmoya) [inline document_ftp/ftpserver/__init__.py:29]: ftp_server > ftpserver, or rename the file

File: document_ftp/wizard/ftp_browse.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv.orm import fields
+from openerp.osv.orm import TransientModel
```
Comment (angelmoya) [inline document_ftp/wizard/ftp_browse.py:25]: Better in new api

**Comment by cmsalmeida** [comment]
Well the migration is working well for us, but to do the PR I will need more time, because all of the PEP8 and new API (which by the way I should congratulate OCA and all of you for this wonderfull job).
I've already fixed some of them. But I'm busy now with internal projects.
You can do the way you think it's better, I will give more contributions as soon I get more time.

File: document_ftp/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+{
+    'name': 'Shared Repositories (FTP)',
+    'version': '8.0.0.0.1',
+    'category': 'Knowledge Management',
+    'author': 'Odoo Community Association (OCA)',
```
Comment (clonedagain) [inline document_ftp/__openerp__.py:29]: The author should contain "Tiny SPRL" as they own the initial copyright.

File: document_ftp/README.rst [hunk 5]
```diff
@@ -0,0 +1,6 @@
+This is a support FTP Interface with document management system.
+================================================================
+
+With this module you would not only be able to access documents through OpenERP
+but you would also be able to connect with them through the file system using the
+FTP client.
```
Comment (clonedagain) [inline document_ftp/README.rst:6]: Please use the OCA template for README - particularly the contributor part is important here, add the initial authors in it + yourself

**Comment by clonedagain** [comment]
It's a pitty the git history is lost, it would have been valuable, particularly to keep authors.
It would be nice to at least document the SHA of the upstream commit from which you forked.

If you need help to make a branch with the full history, just ask, probably one of us can spare a bit of time.

**Comment by hbrunn** [comment]
@cmsalmeida @clonedagain I think history is mandatory, https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0#if-the-module-doesnt-exist-in-the-90-branch has a nice manual on how to do it

**Comment by hbrunn** [comment]
closing due to inactivity, by now, `document_sftp` can take over part of this

**Pull request closed without merging**